app/controllers/renewals_controller.rb

```diff
@@ -78,15 +85,18 @@ def bulk_renewal_summary_flash(response, type)
 
     def renewal_prompt(renewal)
       if renewal.respond_to?(:each)
-        title_text = "#{renewal[0].title}, #{renewal[0].call_number}"
-        non_renewal_reason = error_prompt(renewal[1])
-        title_text + non_renewal_reason
+        renewal_obj, error_message = renewal
+
+        non_renewal_reason = error_prompt error_message
+        renewal_obj.bib_summary + (non_renewal_reason || '')
       else
-        "#{renewal.title}, #{renewal.call_number}"
+        renewal.bib_summary
```

Comment:
😎

Reply:
Oh, and again here, you can turn the `else` into an `end` and remove the `end`

Reply:
This is for to add the title to the message when there are no errors, the successful renewal case.

Reply:
Yup. So, the way I think of it is the top `if` is catching errors. The default scenario is when renewal.respond_to?(:each) ends up being false. and it's the only other scenario, so you can just remove the else...end and just leave it outside the above if that covers errors

Reply:
If I understand correctly, you mean sth like this?

``` 
def renewal_prompt(renewal)
      if renewal.respond_to?(:each)
        renewal_obj, error_message = renewal

        non_renewal_reason = error_prompt error_message
        renewal_obj.bib_summary + (non_renewal_reason || '')
      end
    
      renewal.bib_summary
    end
```
For a successful renewal, `renewal` here only has the checkout obj but for failed ones it is an array of checkout and error msg, without the else in the failed renewal case `renewal.bib_summary` won't work 🤔  If I understand correctly or maybe I am confusing myself here :)

Reply:
refactored this per our conversation on slack